### PR TITLE
Remote ssh & vnc installation

### PR DESCRIPTION
**Pull request by kravciak**
Depend on https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/99

### CODE REVIEW

File: products/sle/main.pm [hunk 1]
```diff
@@ -72,9 +72,9 @@ sub cleanup_needles() {
     remove_desktop_needles("gnome");
     remove_desktop_needles("xfce");
     remove_desktop_needles("minimalx");
-    remove_desktop_needles("textmode");
+    remove_desktop_needles("textmode") unless check_var("REMOTE_MASTER", "ssh");
```
Comment (coolo) [inline products/sle/main.pm]: a) use a proper if/unless - i.e. stick to the coding style of the rest of the file
b) I would rather define VIDEOMODE=text in your test and use that one here. The relationship to REMOTE_* feature is a bit too indirect IMO

File: products/sle/main.pm [hunk 2]
```diff
@@ -72,9 +72,9 @@ sub cleanup_needles() {
...
 
-    if (!check_var("VIDEOMODE", "text")) {
+    if (!check_var("VIDEOMODE", "text") && !check_var("REMOTE_MASTER", "ssh")) {
```
Comment (coolo) [inline products/sle/main.pm]: same here

File: products/sle/main.pm [hunk 3]
```diff
@@ -469,7 +469,7 @@ sub load_inst_tests() {
     }
     if (noupdatestep_is_applicable) {
         loadtest "installation/installer_timezone.pm";
-        loadtest "installation/logpackages.pm";
+        loadtest "installation/logpackages.pm" unless get_var("REMOTE_MASTER");
```
Comment (coolo) [inline products/sle/main.pm]: proper if

File: tests/installation/bootloader.pm [hunk 4]
```diff
@@ -134,6 +135,11 @@ sub run() {
             # ZYPP... enables proxy caching
         }
 
+        if (my $remote = get_var("REMOTE_SLAVE")) {
```
Comment (coolo) [inline tests/installation/bootloader.pm]: please split the assignment and the if

File: tests/installation/bootloader.pm [hunk 5]
```diff
@@ -134,6 +135,11 @@ sub run() {
...
+            type_string_slow " " . get_var("NETSETUP") if get_var("NETSETUP");
+            type_string_slow " nameserver=" . join(",", @{get_host_resolv_conf()->{nameserver}});
```
Comment (coolo) [inline tests/installation/bootloader.pm]: please split this also into 2 lines. The perl style we're trying to use in tests is supposed to be readable to intermediate perl users

File: tests/remote/remote_master.pm [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        select_console 'user-console';
+        type_string "ssh root\@$slave_ip\n";
+        sleep 2;
+        type_string "yes\n";
+        sleep 2;
+        type_string "$password\n";
+        sleep 2;
+        type_string "yast.ssh\n";
+
+        set_var("VIDEOMODE", "text");
```
Comment (coolo) [inline tests/remote/remote_master.pm]: see? this set_var I would move to main.pm and/or the test_suite directly and check for *that* in the needle unregistration

**Comment by kravciak** [comment]
Updated.

**Comment by sysrich** [comment]
Please rebase, if you don't I intend to do it for you but not right now ;)

**Comment by kravciak** [comment]
Already rebased, but found race condition - fixing it now

**Comment by kravciak** [comment]
fixed

**Comment by sysrich** [comment]
Merging, can @dzedro or @Soulofdestiny please setup the scenarios and put them in the Acceptance job group?

**Comment by dzedro** [comment]
@mkravec will do it :)

**Pull request merged**